x/pool-incentives/keeper/grpc_query.go

```diff
@@ -157,6 +158,45 @@ func (q Querier) IncentivizedPools(ctx context.Context, _ *types.QueryIncentiviz
 		}
 	}
 
+	// Retrieve the migration records between balancer pools and concentrated liquidity pools.
+	// This comes from the superfluid keeper, since superfluid is the only pool incentives connected
+	// module that has access to the gamm modules store.
+	migrationRecords, err := q.superfluidKeeper.GetAllMigrationInfo(sdkCtx)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	// Iterate over all migration records.
+	for _, record := range migrationRecords.BalancerToConcentratedPoolLinks {
+		linkedClPoolIsIncentivized := false
+		var duration time.Duration
+
+		// If the linked concentrated liquidity pool is in the list of incentivized pools,
+		// note this and add its balancer counterpart to the list of incentivized pools.
+		for _, incentivizedPool := range incentivizedPools {
+			if incentivizedPool.PoolId == record.ClPoolId {
+				linkedClPoolIsIncentivized = true
+				duration = incentivizedPool.LockableDuration
+				continue
+			}
+		}
+
+		if !linkedClPoolIsIncentivized {
+			continue
+		}
+
+		gaugeId, err := q.Keeper.GetPoolGaugeId(sdkCtx, record.BalancerPoolId, duration)
+		if err == nil {
+			incentivizedPool := types.IncentivizedPool{
+				PoolId:           record.BalancerPoolId,
```

Comment:
In every case the balancer pool will be the pool that is getting indirectly incentivized. This means that there is never a case in which a CL pool does not appear in the incentivized pools list that should be there. That is why we only retrieve the balancer pools at the end, since they are the ones that are getting replaced and funded via the CL gauge